script/lazytox.py

```diff
@@ -120,7 +120,7 @@ def validate_requirements_ok():
 
 async def flake8(files):
     """Exec flake8."""
-    _, log = await async_exec("flake8", "--doctests", *files)
+    _, log = await async_exec("pre-commit", "run", "flake8", "--files", *files)
```

Comment:
Does this require the pre-commit config installed or just the pre-commit executable installed? I think we should keep pre-commit config install optional.

Reply:
I thought it was _not_ optional. And if it was my call, it wouldn't be.

Reply:
So far it's been optional.

Reply:
Anyway it doesn't require having run `pre-commit install`, if that's what you're asking. The config comes in the source tree and `pre-commit run --files ...` uses it directly, no separate install step needed.

Reply:
Just so I understand, what makes something optional or not? The dev environment instructions at https://developers.home-assistant.io/docs/en/development_environment.html tell to run `script/setup` which both installs pre-commit exeuctable and configures it. And I didn't read that step being optional.

Sure, people who know what they're doing can treat _all_ instructions as optional :)

Reply:
I don't think we currently have a contributor facing script that depends on pre-commit config being installed, so in that sense it has been optional. We also said it would be a recommendation when discussing to add pre-commit.

https://github.com/home-assistant/architecture/issues/249

Reply:
Ok, anyway if by pre-commit config installed you mean the git commit hook installed (as in `pre-commit install`), then as mentioned this change doesn't introduce that requirement either.

Reply:
Yes, that's what I mean. :+1: